metricbeat/helper/metricset.go

```diff
@@ -17,6 +17,7 @@ type MetricSet struct {
 // Creates a new MetricSet
 func NewMetricSet(name string, new func() MetricSeter, module *Module) *MetricSet {
 	metricSeter := new()
+	metricSeter.Setup(module.cfg)
```

Comment:
We should probably check for an error here. The consequence is, that NewMetricSet should also be able to return an error. I can add that later for you.